src/app/modules/myinfo/myinfo.service.ts

```diff
@@ -270,6 +280,61 @@ export class MyInfoServiceClass {
       const { fieldValue, isReadOnly } = myInfoData.getFieldValueForAttr(
         myInfoAttr as InternalAttr,
       )
+
+      // Add logging to check if Myinfo field value exists in our constants lists
+      const isFieldValueInMyinfoListCheck = (
+        fieldValue: string,
+        myInfoAttr: string | string[],
+        myInfoList: string[],
+      ) => {
+        const isFieldValueInMyinfoList = myInfoList.includes(fieldValue)
+        if (!isFieldValueInMyinfoList) {
+          logger.error({
+            message: 'Myinfo field value not found in existing list',
+            meta: {
+              action: 'prefillAndSaveMyInfoFields',
+              myInfoFieldValue: fieldValue,
+              myInfoAttr,
+            },
+          })
+        }
+      }
+
+      // Check if field value exists in our constants lists
+      if (fieldValue) {
```

Comment:
Also, was just wondering if we could do this better... like maybe a key value map of myinfoattr -> myinfolist

hence we just query the key value mapping instead of doing this long chain of checks

Reply:
done here! [83cb27a](https://github.com/opengovsg/FormSG/pull/6694/commits/83cb27af4977997df567bec80d98593cf73c1f54)